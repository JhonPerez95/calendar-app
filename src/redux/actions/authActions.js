import { fetchNoToken, fetchToken } from '../../helpers/fetchApi';
import { types } from '../types/types';
import Swal from 'sweetalert2';

export const startLogin = (email, password) => {
  return async (dispatch) => {
    const res = await fetchNoToken('auth', { email, password }, 'POST');
    const body = await res.json();

    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init', new Date().getTime());
      dispatch(login(body.user));
    } else {
      Swal.fire('Error!', body.msg, 'error');
    }
  };
};

export const startRegister = (name, email, password) => {
  return async (dispatch) => {
    const res = await fetchNoToken(
      'auth/new',
      { name, email, password },
      'POST'
    );
    const body = await res.json();

    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init', new Date().getTime());
      dispatch(register(body.user));
    } else {
      Swal.fire('Error!', body.msg, 'error');
    }
  };
};

export const startCheking = () => {
  return async (dispatch) => {
    const res = await fetchToken('auth/renew');
    const body = await res.json();

    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init', new Date().getTime());
      dispatch(login(body.user));
    } else {
      dispatch(checking());
    }
  };
};

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();
    dispatch(logout());
  };
};

const login = (user) => {
  return {
    type: types.authLogin,
    payload: user,
  };
};

const register = (user) => {
  return {
    type: types.authStartRegister,
    payload: user,
  };
};

const checking = () => ({
  type: types.authCheckingFinish,
});

const logout = () => {
  return {
    type: types.authLogout,
  };
};
