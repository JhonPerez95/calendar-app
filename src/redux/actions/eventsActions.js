import { types } from '../types/types';
import { fetchToken } from '../../helpers/fetchApi';
import Swal from 'sweetalert2';
import { prepareEvents } from '../../helpers/prepareEvents';

const eventAddNew = (event) => {
  return { type: types.eventAddNew, payload: event };
};

export const eventSetActive = (event) => {
  return { type: types.eventSetActive, payload: event };
};

export const eventClearActive = () => {
  return { type: types.eventClearActive };
};

export const eventUpdated = (event) => {
  return {
    type: types.eventUpdate,
    payload: event,
  };
};

export const eventDeleted = () => {
  return {
    type: types.eventDeleted,
  };
};

export const eventStartAddEvent = (evento) => {
  return async (dispatch, getState) => {
    const { uid, name } = getState().auth;

    try {
      const res = await fetchToken('event/', evento, 'POST');
      const body = await res.json();

      if (body.ok) {
        evento = { ...body.data, user: { uid, name } };
        dispatch(eventAddNew(evento));
      } else {
        console.log(body);
        Swal.fire('Error!!', 'Valida si ingreso todos los campos', 'error');
      }
    } catch (error) {
      console.log(error);
      Swal.fire('Error!!', error, 'error');
    }
  };
};

export const eventStartLoading = () => {
  return async (dispatch) => {
    try {
      const res = await fetchToken('event');
      const body = await res.json();

      if (body.ok) {
        const e = prepareEvents(body.events);
        dispatch(eventLoader(e));
      } else {
        Swal.fire('Error!', 'Error al consultar en la base de datos', 'error');
      }
    } catch (error) {
      console.log(error);
    }
  };
};

const eventLoader = (events) => {
  return {
    type: types.eventLoader,
    payload: events,
  };
};
