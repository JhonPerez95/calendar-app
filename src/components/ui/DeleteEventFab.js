import React from 'react';
import { eventDeleted } from '../../redux/actions/eventsActions';
import { useDispatch } from 'react-redux';

const DeleteEventFab = () => {
  const dispatch = useDispatch();
  const handleDeleted = () => {
    dispatch(eventDeleted());
  };
  return (
    <button className="btn btn-danger fab-danger" onClick={handleDeleted}>
      <i className="fas fa-trash "></i>
      <span> Borrar</span>
    </button>
  );
};

export default DeleteEventFab;
