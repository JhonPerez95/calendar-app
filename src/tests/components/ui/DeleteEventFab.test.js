import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import DeleteEventFab from '../../../components/ui/DeleteEventFab';
import { Provider } from 'react-redux';
import { eventStartDeleted } from '../../../redux/actions/eventsActions';

// Mock Fuctions
jest.mock('../../../redux/actions/eventsActions', () => ({
  eventStartDeleted: jest.fn(),
}));

// Config Store Redux
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

describe('Test the component ui-DeleteEventFab', () => {
  const wrapper = mount(
    <Provider store={store}>
      <DeleteEventFab />
    </Provider>
  );

  test('should render correct', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call actions dispatch', () => {
    wrapper.find('button').prop('onClick')();

    expect(eventStartDeleted).toHaveBeenCalled();
    expect(eventStartDeleted).toHaveBeenCalledWith();
  });
});
