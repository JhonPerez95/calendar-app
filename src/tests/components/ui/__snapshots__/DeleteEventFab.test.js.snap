// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the component ui-DeleteEventFab should render correct 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash "
  />
  <span>
     Borrar
  </span>
</button>
`;
